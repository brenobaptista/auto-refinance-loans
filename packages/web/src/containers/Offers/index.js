import React, { useState } from "react";
import { Link } from "react-router-dom";
import Offer from "../../components/Offer";
import styles from "./index.module.css";

const Offers = ({ match, history }) => {
  const [loanId, setLoanId] = useState("");

  const handleRefinanceSubmission = (event) => {
    event.preventDefault();

    console.log(loanId);

    loanId !== "" && history.push("/congratulations");
  };

  return (
    <div className={styles.offers}>
      <div className={styles.wrapper}>
        <Link to="/">
          <svg
            xmlns="http://www.w3.org/2000/svg"
            width="24"
            height="24"
            fill="none"
            viewBox="0 0 24 24"
            stroke="#000000"
          >
            <path
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              d="M11 17l-5-5m0 0l5-5m-5 5h12"
            />
          </svg>
        </Link>
        <div className={styles.title}>
          <span className={styles.textImportant}>Select a new loan</span> for
          the Vehicle with ID {match.params.loanId}
        </div>
        <form onSubmit={handleRefinanceSubmission}>
          <Offer
            discount="$246"
            loanId="01"
            setLoanId={setLoanId}
            annualPercentage="2.49%"
            monthlyPayment="$243.00"
            timeRemaining="84 mo"
          />
          <Offer
            discount="$209"
            loanId="02"
            setLoanId={setLoanId}
            annualPercentage="2.49%"
            monthlyPayment="$280.00"
            timeRemaining="72 mo"
          />
          <Offer
            discount="$172"
            loanId="03"
            setLoanId={setLoanId}
            annualPercentage="2.49%"
            monthlyPayment="$317.00"
            timeRemaining="60 mo"
          />
          <button type="submit" className={styles.btn}>
            Finish
          </button>
        </form>
      </div>
    </div>
  );
};

export default Offers;
