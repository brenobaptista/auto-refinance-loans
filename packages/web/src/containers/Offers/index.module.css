@keyframes roll-right {
  0% {
    opacity: 0;
    transform: translate(-16px);
  }

  100% {
    opacity: 1;
    transform: translate(0);
  }
}

.offers {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0 24px;
  height: 100vh;
  color: #6b7280;
  animation: roll-right 1s;
}

@media (max-width: 1024px) {
  .offers {
    display: block;
    height: 100%;
    padding: 32px 24px;
  }
}

.offersWrapper {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  align-items: center;
  gap: 32px;
}

@media (max-width: 1024px) {
  .offersWrapper {
    flex-direction: column;
  }
}

.titleWrapper {
  display: flex;
  justify-content: center;
}

.title {
  max-width: 480px;
  margin: 16px 0 48px;
  font-size: 1.4rem;
  line-height: 1.4;
}

.textImportant {
  font-weight: 600;
  color: #000000;
}

.btn {
  max-width: 300px;
  width: 100%;
  margin-top: 48px;
  padding: 16px 32px;
  border: none;
  border-radius: 25px;
  background-color: #808080;
  color: #ffffff;
  text-align: center;
  font-size: inherit;
  font-weight: 600;
  cursor: pointer;
}

.btnWrapper {
  text-align: center;
}
