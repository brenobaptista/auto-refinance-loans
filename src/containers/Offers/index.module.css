@keyframes roll-right {
  0% {
    opacity: 0;
    transform: translate(-16px);
  }

  100% {
    opacity: 1;
    transform: translate(0);
  }
}

.offers {
  display: flex;
  justify-content: center;
  padding: 32px 24px;
  color: #6b7280;
  animation: roll-right 1s;
}

.title {
  margin-top: 16px;
  font-size: 1.4rem;
}

.textImportant {
  font-weight: 600;
  color: #000000;
}

.btn {
  width: 100%;
  margin: 32px 0 16px;
  padding: 16px 32px;
  border: none;
  border-radius: 25px;
  background-color: #808080;
  color: #ffffff;
  text-align: center;
  font-size: inherit;
  font-weight: 600;
  cursor: pointer;
}
