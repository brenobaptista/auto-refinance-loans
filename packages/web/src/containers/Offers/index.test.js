import { render, fireEvent } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import Offers from './index'

beforeEach(() => {
  fetch.resetMocks()
})

test('matches snapshot', async () => {
  fetch.mockResponseOnce(
    JSON.stringify({
      id: 1,
      lender: 'The Big Bank',
      annualPercentage: 0.142,
      loanBalance: 15000,
      monthlyPayment: 377.5,
      paidPeriod: 18,
      totalPeriod: 72,
      vehicle: {
        photo: '1.jpg',
        make: 'HONDA',
        model: 'ACCORD',
        year: 2019,
        mileage: 41855
      }
    })
  )

  const { asFragment, findByRole, getByRole } = render(
    <Offers match={{ params: { loanId: 1 } }} />,
    {
      wrapper: MemoryRouter
    }
  )

  const option = await findByRole('radio', {
    name: 'Lower you payment by 94.74 per month'
  })
  fireEvent.click(option)

  const finish = getByRole('button', { name: 'Finish' })
  fireEvent.click(finish)

  expect(asFragment()).toMatchSnapshot()
})

test('matches snapshot when expection', () => {
  fetch.mockReject(() => Promise.reject('API is down'))

  const { asFragment } = render(<Offers match={{ params: { loanId: 1 } }} />, {
    wrapper: MemoryRouter
  })

  expect(asFragment()).toMatchSnapshot()
})
