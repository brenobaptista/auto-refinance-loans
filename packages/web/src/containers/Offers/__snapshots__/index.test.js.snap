// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="offers"
  >
    <div>
      <div
        class="titleWrapper"
      >
        <div>
          <a
            href="/"
          >
            <svg
              fill="none"
              height="24"
              stroke="#000000"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11 17l-5-5m0 0l5-5m-5 5h12"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </a>
          <div
            class="title"
          >
            <span
              class="textImportant"
            >
              Select a new loan
            </span>
             for the 2019 HONDA ACCORD
          </div>
        </div>
      </div>
      <form>
        <div
          class="offersWrapper"
        >
          <div
            class="offer"
          >
            <label
              class="loanInfo"
            >
              <span
                class="discount textImportant"
              >
                Lower you payment by 94.74 per month
              </span>
              <input
                class="radio"
                name="offer"
                type="radio"
              />
            </label>
            <div
              class="loanInfo"
            >
              <div>
                APR
              </div>
              <div
                class="textImportant"
              >
                7.10%
              </div>
            </div>
            <hr />
            <div
              class="loanInfo"
            >
              <div>
                Monthly payment
              </div>
              <div
                class="textImportant"
              >
                $282.76
              </div>
            </div>
            <hr />
            <div
              class="loanInfo"
            >
              <div>
                Time remaining
              </div>
              <div
                class="textImportant"
              >
                84 mo
              </div>
            </div>
          </div>
          <div
            class="offer"
          >
            <label
              class="loanInfo"
            >
              <span
                class="discount textImportant"
              >
                Lower you payment by 66.81 per month
              </span>
              <input
                class="radio"
                name="offer"
                type="radio"
              />
            </label>
            <div
              class="loanInfo"
            >
              <div>
                APR
              </div>
              <div
                class="textImportant"
              >
                7.10%
              </div>
            </div>
            <hr />
            <div
              class="loanInfo"
            >
              <div>
                Monthly payment
              </div>
              <div
                class="textImportant"
              >
                $310.69
              </div>
            </div>
            <hr />
            <div
              class="loanInfo"
            >
              <div>
                Time remaining
              </div>
              <div
                class="textImportant"
              >
                72 mo
              </div>
            </div>
          </div>
          <div
            class="offer"
          >
            <label
              class="loanInfo"
            >
              <span
                class="discount textImportant"
              >
                Lower you payment by 26.92 per month
              </span>
              <input
                class="radio"
                name="offer"
                type="radio"
              />
            </label>
            <div
              class="loanInfo"
            >
              <div>
                APR
              </div>
              <div
                class="textImportant"
              >
                7.10%
              </div>
            </div>
            <hr />
            <div
              class="loanInfo"
            >
              <div>
                Monthly payment
              </div>
              <div
                class="textImportant"
              >
                $350.58
              </div>
            </div>
            <hr />
            <div
              class="loanInfo"
            >
              <div>
                Time remaining
              </div>
              <div
                class="textImportant"
              >
                60 mo
              </div>
            </div>
          </div>
        </div>
        <div
          class="btnWrapper"
        >
          <button
            class="btn"
            type="submit"
          >
            Finish
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when expection 1`] = `<DocumentFragment />`;
